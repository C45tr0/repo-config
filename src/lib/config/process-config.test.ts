import processConfig from "./process-config";

test("throws error when files conflict", () => {
  expect(() => {
    processConfig({
      extensions: {
        "node_modules/extension1": {
          files: {
            "tsconfig.json": "node_modules/extension1/files/tsconfig.json"
          }
        },
        "node_modules/extension2": {
          files: {
            "tsconfig.json": "node_modules/extension2/files/tsconfig.json"
          }
        }
      }
    });
  }).toThrowErrorMatchingSnapshot();
});

test("allow config to remove file", () => {
  expect(
    processConfig({
      extensions: {
        "node_modules/extension1": {
          files: {
            "tsconfig.json": false
          }
        },
        "node_modules/extension2": {
          files: {
            "tsconfig.json": "node_modules/extension2/files/tsconfig.json"
          }
        }
      }
    })
  ).toMatchSnapshot();
});

test("processes extension configs", () => {});
