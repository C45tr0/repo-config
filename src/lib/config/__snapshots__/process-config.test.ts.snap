// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allow config to remove file 1`] = `
Object {
  "cmds": Object {},
  "files": Object {
    "tsconfig.json": false,
  },
}
`;

exports[`throws error when files conflict 1`] = `"Two or more configs are trying to copy the same file: 'tsconfig.json' from source 1: 'node_modules/extension1/files/tsconfig.json' and source 2: 'node_modules/extension2/files/tsconfig.json'"`;
